transport/src/main/java/io/camunda/zeebe/transport/stream/impl/messages/MessageUtil.java

```diff
@@ -34,4 +34,11 @@ private static <R extends BufferReader> R parseRequest(final byte[] bytes, final
 
     return request;
   }
+
+  public static byte[] encodeResponse(final StreamResponse response) {
+    final var buffer = new byte[response.getLength()];
+    response.write(new UnsafeBuffer(buffer), 0);
+
+    return buffer;
+  }
```

Comment:
🔧 💭 Don't we have some utility for such, which just takes a BufferWriter and writes to a buffer/array?